src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1SpaceSearchTest.kt

```diff
@@ -103,4 +347,217 @@ class Cas1SpaceSearchTest : InitialiseDatabasePerClassTestBase() {
+        withCharacteristicsList(
+          listOf(
+            Cas1SpaceCharacteristic.entries.first { it != characteristic }.asCharacteristicEntity(),
+          ),
+        )
+      }
+
+      unexpectedPremises.forEach {
+        roomEntityFactory.produceAndPersist {
+          withPremises(it)
+          withCharacteristicsList(
+            listOf(
+              Cas1SpaceCharacteristic.entries.first { it != characteristic }.asCharacteristicEntity(),
+            ),
+          )
+        }
+      }
+
+      val searchParameters = Cas1SpaceSearchParameters(
+        startDate = LocalDate.now(),
+        durationInDays = 14,
+        targetPostcodeDistrict = "SE1",
+        requirements = Cas1SpaceSearchRequirements(
+          apTypes = null,
+          spaceCharacteristics = listOf(characteristic),
+          genders = null,
+        ),
+      )
+
+      val response = webTestClient.post()
+        .uri("/cas1/spaces/search")
+        .header("Authorization", "Bearer $jwt")
+        .bodyValue(searchParameters)
+        .exchange()
+        .expectStatus()
+        .isOk
+        .returnResult(Cas1SpaceSearchResults::class.java)
+
+      val results = response.responseBody.blockFirst()!!
+
+      assertThat(results.resultsCount).isEqualTo(5)
+      assertThat(results.searchCriteria).isEqualTo(searchParameters)
+
+      assertThatResultMatches(results.results[0], expectedPremises[0])
+      assertThatResultMatches(results.results[1], expectedPremises[1])
+      assertThatResultMatches(results.results[2], expectedPremises[2])
+      assertThatResultMatches(results.results[3], expectedPremises[3])
+      assertThatResultMatches(results.results[4], expectedPremises[4])
+    }
+  }
+
+  @Test
+  fun `Filtering APs by multiple characteristics only returns APs with all characteristics`() {
+    postCodeDistrictFactory.produceAndPersist {
+      withOutcode("SE1")
+      withLatitude(-0.07)
+      withLongitude(51.48)
+    }
+
+    `Given a User` { _, jwt ->
+      val expectedPremises = approvedPremisesEntityFactory.produceAndPersistMultipleIndexed(5) {
+        withYieldedProbationRegion {
+          probationRegionEntityFactory.produceAndPersist {
+            withYieldedApArea {
+              apAreaEntityFactory.produceAndPersist()
+            }
+          }
+        }
+        withYieldedLocalAuthorityArea {
+          localAuthorityEntityFactory.produceAndPersist()
+        }
+        withLatitude((it * -0.01) - 0.08)
+        withLongitude((it * 0.01) + 51.49)
+        withCharacteristicsList(Cas1SpaceCharacteristic.entries.slice(1..3).map { it.asCharacteristicEntity() })
+      }
+
+      expectedPremises.forEach {
+        roomEntityFactory.produceAndPersist {
+          withPremises(it)
+          withCharacteristicsList(Cas1SpaceCharacteristic.entries.slice(1..3).map { it.asCharacteristicEntity() })
+        }
+      }
+
+      println("===\nPremises:\n${expectedPremises.map { it.id to it.characteristics }}\n===")
```

Comment:
see previous comment

Reply:
This has been removed as it's no longer necessary.